### ISSUE TITLE
Compilation errors that seem to be related to git2

### ISSUE THREAD

**Issue by ubolonton** [emacs-ng/emacs-ng#111]
I'm getting these errors compiling from 08fbf4ad

## Linux

```
./configure \
    --with-modules \
    --with-threads \
    --without-nativecomp \
    --without-gnutls \
    --without-imagemagick \
    --without-x \
    --without-dbus \
    --without-makeinfo
```

<details>

<summary>Error</summary>

```
   Compiling git2 v0.13.15
warning: unused variable: `repo`
  --> src/git.rs:12:12
   |
12 |         Ok(repo) => path,
   |            ^^^^ help: if this is intentional, prefix it with an underscore: `_repo`
   |
   = note: `#[warn(unused_variables)]` on by default

error: could not compile `emacsng`

Caused by:
  process didn't exit successfully: `rustc --crate-name emacsng --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type staticlib --emit=dep-info,link -C opt-level=z -C panic=abort -C lto -C codegen-units=1 --cfg 'feature="default"' --cfg 'feature="use-xml2"' -C metadata=55c5b5e2a2079b62 -C extra-filename=-55c5b5e2a2079b62 --out-dir /home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps -L dependency=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps --extern cfg_if=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libcfg_if-9529c48ef2953a20.rlib --extern crossbeam=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libcrossbeam-9e5f9fdf61b85fb5.rlib --extern deno=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libdeno-11d8ae5411ea93e5.rlib --extern deno_core=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libdeno_core-c8e2ec37359ca618.rlib --extern deno_runtime=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libdeno_runtime-8b2103f113d89108.rlib --extern errno=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liberrno-9e6f3b41308d76fe.rlib --extern field_offset=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libfield_offset-95d529123245967e.rlib --extern flate2=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libflate2-0d81cf2203cce24f.rlib --extern futures=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libfutures-226ecede1356f2ad.rlib --extern git2=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libgit2-26203f3190bf6b11.rlib --extern itertools=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libitertools-0eb5f6e7389d0c9d.rlib --extern lazy_static=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblazy_static-78cceb195b642a98.rlib --extern libc=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblibc-ab503b444252b559.rlib --extern line_wrap=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libline_wrap-028d8f284f577f72.rlib --extern lisp=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblisp-e4718c944e9275ae.rlib --extern lisp_macros=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblisp_macros-fd4dbf916713cbd0.so --extern lisp_util=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblisp_util-095caf9862df1e95.rlib --extern lsp_server=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/liblsp_server-0ecef63603c4b374.rlib --extern openssl=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libopenssl-ff87d105f67835d0.rlib --extern openssl_sys=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libopenssl_sys-6c6a98dc51f7423e.rlib --extern rand=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/librand-183e0da0a3ce9260.rlib --extern remacs_lib=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libremacs_lib-bbd2ca18e959bbbc.rlib --extern rusty_v8=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/librusty_v8-71521dfac9ac4675.rlib --extern serde_json=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libserde_json-8d844871f5395f22.rlib --extern systemstat=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libsystemstat-06ed65eef74c01cb.rlib --extern tokio=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libtokio-cde8b4fbe3b373fe.rlib --extern tokio_rustls=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/deps/libtokio_rustls-c37a7b957e4ea0a4.rlib -L /home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/gn_out/obj -L native=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/build/ring-49998ceb61880dbc/out -L native=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/build/sys-info-5ae78d697386a41b/out -L native=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/build/libgit2-sys-3f234e19096dee3b/out/build -L native=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/build/libssh2-sys-fa4857cc93638a2c/out/build -L native=/home/ubolonton/Programming/Tools/emacs/emacs-ng/rust_src/target/release/build/openssl-sys-f4a4fba5abb8e92d/out/openssl-build/install/lib` (signal: 9, SIGKILL: kill)
Makefile:696: recipe for target '../rust_src/target/release/libemacsng.a' failed
make[1]: *** [../rust_src/target/release/libemacsng.a] Error 101
make[1]: Leaving directory '/home/ubolonton/Programming/Tools/emacs/emacs-ng/src'
Makefile:445: recipe for target 'src' failed
make: *** [src] Error 2

```

</details>

## macOS

```
./configure \
    --with-modules \
    --with-threads \
    --with-ns \
    --with-imagemagick \
    --with-gnutls=no \
    --without-x \
    --without-dbus \
    --without-makeinfo
```

<details>

<summary>Error</summary>

```
make[2]: Leaving directory '/Users/ubolonton/Programming/Tools/emacs/emacs-ng/admin/unidata'
gcc -o temacs.tmp \
  -Demacs  -I. -I. -I../lib -I../lib      -D_REENTRANT -I/opt/local/include/librsvg-2.0 -I/opt/local/include/gdk-pixbuf-2.0 -I/opt/local/include/cairo -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include/pixman-1 -I/opt/local/include/ossp -I/opt/local/include/freetype2 -I/opt/local/include/libpng16 -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/opt/local/include/ImageMagick-6 -I/opt/local/include/libpng16 -I/opt/local/include/libxml2               -MMD -MF deps/.d -MP       -pthread -DEMACS_EXTERN_INLINE -Wno-switch -Wno-pointer-sign -Wno-string-plus-int -Wno-unknown-attributes -Wno-initializer-overrides -Wno-tautological-compare -Wno-tautological-constant-out-of-range-compare -g3 -O2  -framework AppKit -framework IOKit -framework Carbon -framework IOSurface -L.././rust_src/target/release -framework Security \
    dispnew.o frame.o scroll.o xdisp.o menu.o  window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o    emacs.o keyboard.o macros.o keymap.o sysdep.o bignum.o buffer.o filelock.o insdel.o marker.o minibuf.o fileio.o dired.o cmds.o casetab.o casefiddle.o indent.o search.o regex-emacs.o undo.o alloc.o pdumper.o data.o doc.o editfns.o callint.o eval.o floatfns.o fns.o font.o print.o lread.o emacs-module.o syntax.o  bytecode.o comp.o dynlib.o process.o gnutls.o callproc.o region-cache.o sound.o timefns.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o kqueue.o  profiler.o decompress.o thread.o systhread.o        fontset.o fringe.o image.o  json.o  nsterm.o nsfns.o nsmenu.o nsselect.o nsimage.o macfont.o -lemacsng terminfo.o lastfile.o      ../lib/libgnu.a       -ltiff -ljpeg -L/opt/local/lib -lpng16 -lgif    -L/opt/local/lib -lrsvg-2 -lm -lgio-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -lintl -Wl,-framework -Wl,CoreFoundation -lcairo -L/opt/local/lib -lMagickWand-6.Q16 -lMagickCore-6.Q16            -L/opt/local/lib -lxml2     -lncurses            -L/opt/local/lib -llcms2   -lz   -L/opt/local/lib -ljansson -lgmp  -ldl -lm -lresolv
Undefined symbols for architecture x86_64:
  "_libiconv", referenced from:
      _git_path_iconv in libemacsng.a(path.o)
  "_libiconv_close", referenced from:
      _git_path_iconv_clear in libemacsng.a(path.o)
  "_libiconv_open", referenced from:
      _git_path_direach in libemacsng.a(path.o)
      _git_path_iconv_init_precompose in libemacsng.a(path.o)
      _git_path_diriter_init in libemacsng.a(path.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:745: temacs] Error 1
make[1]: Leaving directory '/Users/ubolonton/Programming/Tools/emacs/emacs-ng/src'
make: *** [Makefile:445: src] Error 2
```

</details>

The error on macOS seems to be that the `make` rules are not aware of the lib/include paths used by `cargo`?

**Comment by DavidDeSimone** [comment]
This is really strange to me due to the fact that CI is building fine, in addition to my personal machine. 
The linux case is especially confusing, because I can see that it is trying to link against the git rlib. I also don't understand the `(signal: 9, SIGKILL: kill)`

**Comment by DavidDeSimone** [comment]
Dumb question- Do you have git already installed on either of these boxes?

**Comment by brotzeit** [comment]
Maybe we should just remove it for now since it's not really used and I'm currently trying to get things working with libegit.

**Comment by DavidDeSimone** [comment]
@brotzeit I don't think it's a bad idea to disable the feature until we can reproduce this or identify a fix.

**Comment by ubolonton** [comment]
>  The linux case is especially confusing, because I can see that it is trying to link against the git rlib. I also don't understand the `(signal: 9, SIGKILL: kill)`

Yeah, that's pretty weird. I can build other (pure Rust) projects that use git2-rs.

**Comment by DavidDeSimone** [comment]
I actually vote to close this and just track this work under the general libggit2 issue - I don't want people to think that builds are broken.

**Comment by ubolonton** [comment]
The Linux one was just my machine running out of memory.

The macOS one is fixed by #111, though I got another issue, with generating `bindings.rs`. Will discuss more on gitter.
